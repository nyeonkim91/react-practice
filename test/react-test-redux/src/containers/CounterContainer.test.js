import React from 'react';
import { mount } from 'enzyme';
import CounterContainer from './CounterContainer';
import configureMockStore from 'redux-mock-store';
import * as counterActions from '../store/modules/counter';

describe('CounterContainer', () => {
  let component = null;
  const mockStore = configureMockStore();

  // 데이터들을 받아올 가짜 스토어 만들기
  let store = mockStore({
    counter: {
      number: 0
    }
  });

  it('renders properly', () => {
    const context = { store };
    component = mount(<CounterContainer />, { context });
    // OR component = mount(<CounterContainer store={store} />);
  })

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  })

  it('dispatches INCREASE action', () => {
    // 스토어에 액션이 디스패치 되면
    // 디스패치된 액션들의 목록을 store.getActions 를 통해 조회 가능
    component.find('button').at(0).simulate('click');
    expect(store.getActions()[0]).toEqual(counterActions.increase());
  });

  it('dispatches DECREASE action', () => {
    component.find('button').at(1).simulate('click');
    expect(store.getActions()[1]).toEqual(counterActions.decrease());
  });
})
