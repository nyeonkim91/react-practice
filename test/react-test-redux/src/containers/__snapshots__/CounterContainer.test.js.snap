// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterContainer matches snapshot 1`] = `
<Connect(CounterContainer)>
  <CounterContainer
    CounterActions={
      Object {
        "decrease": [Function],
        "default": [Function],
        "increase": [Function],
      }
    }
    number={0}
  >
    <Counter
      onDecrease={[Function]}
      onIncrease={[Function]}
      value={0}
    >
      <div>
        <h1>
          카운터
        </h1>
        <h2>
          0
        </h2>
        <button
          onClick={[Function]}
        >
          +
        </button>
        <button
          onClick={[Function]}
        >
          -
        </button>
      </div>
    </Counter>
  </CounterContainer>
</Connect(CounterContainer)>
`;
